### DISCUSSION TITLE
Next-Auth still shows loading page for a tiny bit of time even though session is authenticated

### DISCUSSION THREAD

**Discussion by kecoliva** [nextauthjs/next-auth#4139] [Help] [answered]
### Environment

System:
    OS: Windows 10 10.0.22000
    CPU: (16) x64 AMD Ryzen 9 5900HX with Radeon Graphics
    Memory: 6.20 GB / 15.40 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (98.0.1108.62)
  npmPackages:
    next: 12.1.0 => 12.1.0
    next-auth: ^4.2.1 => 4.2.1
    react: 17.0.2 => 17.0.2

### Reproduction URL

https://github.com/kecoliva/nextauthjs-loading-issue

### Describe the issue

Even though the session is already authenticated, there will be a brief display of the loading page. Based on the logic of _app.ts, it should no longer render a loading page once the session is already authenticated.

### How to reproduce

1. go to locahost:3000/
2. you will see a brief loading then login page
3. login with (olivakyle@outlook.com:1amAf1sh)
4. it will redirect you to go to locahost:3000/ but now there's a sign out button
5. stay on localhost:3000/ then refresh the page.

### Expected behavior

Even though the session is already authenticated, there will be a brief display of the loading page.

Also I added a console.log at _app.tsx:44 before it renders the login page and as stated in the bug, it will still trigger that console.log even though the session is already authenticated.

**Comment by ThangHuuVu** [comment, marked as answer]
If you want to avoid Flash of Unauthenticated Content (FOUC), please use SSR. 
See: 
- https://nextjs.org/docs/authentication#authentication-patterns
- https://nextjs.org/docs/basic-features/pages#server-side-rendering

**Reply by kecoliva** [reply]
Got it. I misunderstood this paragraph

_Due to the way Next.js handles getServerSideProps / getInitialProps, every protected page load has to make a server-side request to check if the session is valid and then generate the requested page. This alternative solution allows for showing a loading state on the initial check and every page transition afterward will be client-side, without having to check with the server and regenerate pages._

 from next-auth https://next-auth.js.org/getting-started/client#custom-client-session-handling. Do you have any recommendations on how I can make tweak this to use SSR?

**Reply by ThangHuuVu** [reply]
@kecoliva You'll have to fetch the session in `getServerSideProps` using `getServerSession`:

```
import { getServerSession } from "next-auth"
import { authOptions } from [...nextauth].js

...

export async function getServerSideProps(ctx) {
  return {
    props: {
      session: await getServerSession(ctx, authOptions)
    }
  }
}
```
`authOptions` is the configuration object you pass in `NextAuth`.

Note that this is an experimental API. We are still improving/evaluating it. See: https://github.com/nextauthjs/next-auth/pull/4116

**Reply by chrisrhymes** [reply]
Thanks for the example @ThangHuuVu. Is there a way to use getServerSession() in getInitialProps in `_app.tsx`? 

I ask because I have .mdx content pages that don't use getServerSideProps so wanted to extract getting the session to the `_app.tsx` file.

**Reply by ThangHuuVu** [reply]
@chrisrhymes I suppose you can follow this example from our [updated docs](https://next-auth.js.org/getting-started/client#sessionprovider):

```js
import { unstable_getServerSession } from "next-auth/next"
import { authOptions } from './api/auth/[...nextauth]'

...

export async function getServerSideProps({ req, res }) {
  return {
    props: {
      session: await unstable_getServerSession(req, res, authOptions)
    }
  }
}
```

**Comment by litigodv** [comment]
Anyone knows how to implement this ssr strategy on Next.js V14.x ?
